### PR TITLE
Add LivingEntity#swingHand(EquipmentSlot) convenience method

### PR DESCRIPTION
**Pull request by SoSeDiK**
I have dozens of methods where I operate with an item from events solely based on the passed slot.
Old and ugly code (already shortened by not using brackets) would require me to do this:
```java
if (hand == EquipmentSlot.HAND) player.swingMainHand();
else if (hand == EquipmentSlot.OFF_HAND) player.swingOffHand(); // or just else considering no third hand
```
It can now be as simple as `player.swingHand(hand);` :)

### CODE REVIEW

File: patches/api/0399-Add-extra-swing-method-for-convenience.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
++    /**
++     * Makes this entity swing their hand.
++     *
++     * <p>This method does nothing if this entity does not
++     * have an animation for swinging their hand.
++     *
++     * @param hand hand to be swung, either {@link org.bukkit.inventory.EquipmentSlot#HAND} or {@link org.bukkit.inventory.EquipmentSlot#OFF_HAND}
++     */
++    default void swingHand(@NotNull org.bukkit.inventory.EquipmentSlot hand) {
++        if (hand == org.bukkit.inventory.EquipmentSlot.HAND) swingMainHand();
```
Comment (jpenilla) [inline patches/api/0399-Add-extra-swing-method-for-convenience.patch]: add brackets to ifs and `this.` to instance method calls. also should have a preconditions check that one of the required values was passed

**Review by jpenilla — CHANGES_REQUESTED** [review]

File: patches/api/0399-Add-extra-swing-method-for-convenience.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SoSeDiK <mrsosedik@gmail.com>
+Date: Tue, 11 Oct 2022 22:35:56 +0300
+Subject: [PATCH] Add extra swing method for convenience
```
Comment (jpenilla) [inline patches/api/0399-Add-extra-swing-method-for-convenience.patch]:
```suggestion
Subject: [PATCH] Add LivingEntity#swingHand(EquipmentSlot) convenience method
```

File: patches/api/0399-Add-extra-swing-method-for-convenience.patch (hunk 1 above)
Comment (SoSeDiK) [reply patches/api/0399-Add-extra-swing-method-for-convenience.patch]: Done & squished to a single commit.
Oh, forgot about preconditions, oops.

Comment (jpenilla) [reply patches/api/0399-Add-extra-swing-method-for-convenience.patch]: might have missed my edit on the other comment too, sorry :p
(naming the patch with swingHand (not swing) like the method)

Comment (SoSeDiK) [reply patches/api/0399-Add-extra-swing-method-for-convenience.patch]: Oh yea, just copypasted it without noticing 😅
Probably should be sleeping now instead of doing changes at 2 AM :)

**Comment by SoSeDiK** [comment]
By the way, would `EquipmentSlot#isHand` be a neat addition? 🤔
(can do a separate PR or add a commit to this one with separate .patch)
 
You'd have to do a long `EquipmentSlot.HAND || EquipmentSlot.OFF_HAND` instead of simple `#isHand()` to manually check for the hand if needed.

Edit: If so, maybe `#isArmor` could also serve a purpose for checking armor slots? Since `!isHand()` might be off in case of new equipment slots in the future, and manually checking all four is "meh".

File: patches/api/0399-Add-LivingEntity-swingHand-EquipmentSlot-convenience.patch [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
++     * Makes this entity swing their hand.
++     *
++     * <p>This method does nothing if this entity does not
++     * have an animation for swinging their hand.
++     *
++     * @param hand hand to be swung, either {@link org.bukkit.inventory.EquipmentSlot#HAND} or {@link org.bukkit.inventory.EquipmentSlot#OFF_HAND}
++     * @throws IllegalArgumentException if invalid hand is passed
++     */
++    default void swingHand(@NotNull org.bukkit.inventory.EquipmentSlot hand) {
++        com.google.common.base.Preconditions.checkArgument(hand == org.bukkit.inventory.EquipmentSlot.HAND || hand == org.bukkit.inventory.EquipmentSlot.OFF_HAND, "EquipmentSlot was not a hand!");
```
Comment (jpenilla) [inline patches/api/0399-Add-LivingEntity-swingHand-EquipmentSlot-convenience.patch]: Would be nice to include the value of `hand` in the exception message

as far as isHand/isArmor, it seems reasonable to me. probably do it in a separate PR, since with the aforementioned change this one is basically ready

Comment (SoSeDiK) [reply patches/api/0399-Add-LivingEntity-swingHand-EquipmentSlot-convenience.patch]: Hopefully, this will do :)

**Review by jpenilla — APPROVED** [review]
